internal/mode/static/manager.go

```diff
@@ -225,25 +226,51 @@ func StartManager(cfg config.Config) error {
 	objects, objectLists := prepareFirstEventBatchPreparerArgs(cfg.GatewayClassName, cfg.GatewayNsName)
 	firstBatchPreparer := events.NewFirstEventBatchPreparerImpl(mgr.GetCache(), objects, objectLists)
 
+	// Ensure NGINX is running before registering metrics & starting the manager.
+	if err := ngxruntime.EnsureNginxRunning(ctx); err != nil {
+		return fmt.Errorf("NGINX is not running: %w", err)
+	}
+
+	if cfg.MetricsConfig.Enabled {
+		if err := configureNginxMetrics(cfg.GatewayClassName); err != nil {
+			return err
+		}
+	}
+
 	eventLoop := events.NewEventLoop(
 		eventCh,
 		cfg.Logger.WithName("eventLoop"),
 		eventHandler,
-		firstBatchPreparer)
+		firstBatchPreparer,
+	)
 
 	if err = mgr.Add(eventLoop); err != nil {
 		return fmt.Errorf("cannot register event loop: %w", err)
 	}
 
-	// Ensure NGINX is running before registering metrics & starting the manager.
-	if err := ngxruntime.EnsureNginxRunning(ctx); err != nil {
-		return fmt.Errorf("NGINX is not running: %w", err)
-	}
-
-	if cfg.MetricsConfig.Enabled {
-		if err := configureNginxMetrics(cfg.GatewayClassName); err != nil {
+	if cfg.LeaderElection.Enabled {
+		leaderElector, err := newLeaderElector(leaderElectorConfig{
+			kubeConfig: clusterCfg,
+			recorder:   recorder,
+			onStartedLeading: func(ctx context.Context) {
+				logger.Info("Started leading")
+				statusUpdater.WriteLastStatuses(ctx)
+			},
+			onStoppedLeading: func() {
+				logger.Info("Stopped leading")
+			},
+			lockNs:   cfg.Namespace,
+			lockName: cfg.LeaderElection.LockName,
+			identity: cfg.LeaderElection.Identity,
+		})
+		if err != nil {
 			return err
 		}
+
+		statusUpdater.SetLeaderElector(leaderElector)
```

Comment:
Because I need the status updater to construct the leader elector, since the leader elector calls methods on the status updater

Reply:
I see. thanks for clarification. looks like we have a circular dependency.

I still think that because leader elector doesn't change during runtime, we will be better off if we don't allow to change it in the status updater by design.

Perhaps a different design could be used here. Consider the following interface:
```go
// Updater updates statuses of the Gateway API resources.
// Updater can be disabled. In this case, it will stop updating the statuses of resources, while
// always saving the statuses of the last Update call. This is used to support multiple replicas of
// control plane being able to run simultaneously where only the leader will update statuses.
type Updater interface {
	// Update updates the statuses of the resources. If Updater is disabled, it will not update the statuses.
	Update(context.Context, Statuses)
        // Disable disables status updates.
        Disable()
        // Enable enables status updates. The updater will update the statuses in Kubernetes API to ensure they match the
        // statuses of the last Update invocation.
        Enable(context.Context)
}
```

The leader elector will call Enable() or Disable() on the updater. This means Updater will not directly depend on the leader elector.

Reply:
I like this a lot. Thanks. I'll refactor